### PR TITLE
TEC Card Component Internal/External Chip

### PR DESCRIPTION
**Pull request by zkdlinyne**
### Summary 

This mini pull request adds the missing frontend fix to the TEC Card Components. Now, once the toggle is ON, team event cards display a small green/blue (depending on internal/external nature) chip in the top right corner of the card, so members can easily see which team events are internal/external outside of just the dropdown menu. 

### Test Plan 

1. Flip the toggle ON on the admin side
2. Create a new team event and mark it as internal or external
3. Ensure the team event shows up in the dropdown menu on the member side when submitting a TEC
4. Click Submit and scroll down to the TEC Dashboard to see your pending TEC submission. On that card, there should be an internal/external chip in the top right corner. 

Screenshot 2026-09-19 at 8 41 39 PM

### Notes 

Let me know if email text is worth changing as well! I know it was discussed that it was unnecessary, but I think it's a pretty easy fix.

### CODE REVIEW

File: backend/src/API/reimbursementAPI.ts [hunk 1]
```diff
@@ -130,7 +130,8 @@ export const createReimbursementRequest = async (
   request: Partial<ReimbursementRequest>,
   user: IdolMember
 ): Promise<ReimbursementRequest> => {
-  if (!hasReimbursementRequesterPermissions(user.role)) {
+  const roleForCheck = user.email === 'ek782@cornell.edu' ? 'pm' : user.role;
+  if (!hasReimbursementRequesterPermissions(roleForCheck)) {
     throw new PermissionError(
```
Comment (jane-leon) [inline backend/src/API/reimbursementAPI.ts]: i think this is unrelated to the pr, and wasn't meant to be included. please remove if so, and if you want to try testing with that role, try changing manually in firestore under the members storage file.

**Review by jane-leon — CHANGES_REQUESTED** [review]
Hii Eva! good job on this pr, love the layout just left a small fix !

File: backend/src/API/reimbursementAPI.ts (hunk 1 above)
Comment (zkdlinyne) [reply backend/src/API/reimbursementAPI.ts]: Oh yes I was testing and forgot to remove - thanks for letting me know!

**Comment by mshi-27** [comment]
Hi Eva, ran the code with your test plan and everything works as intended!

I also checked a few edge cases:

- Longer event names wrap cleanly without pushing the chip into a weird position or causing overlap.

- Toggle-off state still hides the chip as expected.

- Chip renders correctly for both pending and approved events.

- Events without a kind do not render an empty chip area.

The new chip makes the internal/external status much easier to notice from the dashboard, great work!

**Review by jane-leon — APPROVED** [review]
